src/unix/posix-poll.c

```diff
@@ -115,7 +115,9 @@ static void uv__pollfds_del(uv_loop_t* loop, int fd) {
       loop->poll_fds[loop->poll_fds_used].fd = -1;
       loop->poll_fds[loop->poll_fds_used].events = 0;
       loop->poll_fds[loop->poll_fds_used].revents = 0;
-      return;
+      /* This method is called with an fd of -1 to purge the invalidated fd's, so we may possibly have multiples to remove. */
+      if (-1 != fd)
+        return;
```

Comment:
In the case that we do *not* return, we must avoid incrementing the loop counter before the next iteration.  Otherwise the value just swapped here from the previous end of the array will be skipped.  The loop could be changed to increment `i` only when the `fd` does not match.

Reply:
Great catch!

Reply:
The revised logic LGTM.  Also, thanks for finding this in the first place.